File: merge sorted array.py
# TC-O(n+m)
# SC-O(1)
def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
    """
    Do not return anything, modify nums1 in-place instead.
    """
    # brute force
    # if n!=0:
    #     j=0
    #     for i in range(m,len(nums1)):
    #         nums1[i]=nums2[j]
    #         j=j+1
    #         print(nums1)
    # nums1=nums1.sort()
    # return nums1
    if ( n= =0):
        return i=m-1
    j=n-1
    k=m+n - 1
    while(i>= 0 and j>=0):
        if nums1[i]>nums 2 [j]:
            nums1[k]=nums 1 [i]

            i-=1
        else:
            nums1[k]=nums 2 [j]

            j-=1
        k-=1
    while j>=0:
        nums1[k]=nums 2 [j]
        j-=1
        k-=1

File: MergeSortedArray.js
// Time Complexity : O (m +n) where m and n are the length of two arrays
// Space Complexity : O(1)
// Did this code successfully run on Leetcode : Yes
// Three line explanation of solution in plain english : Using three pointers p1,p2,p3 and looping backwards by replacing zeros with the higher value between nums1 and nums2. Repacing the nums1 in place .

/**
 * @param {number[]} nums1
 * @param {number} m
 * @param {number[]} nums2
 * @param {number} n
 * @return {void} Do not return anything, modify nums1 in-place instead.
 */
var merge = function(nums1, m, nums2, n) {
  var p1 = m - 1;
  var p2 = n - 1;
  var p3 = m + n - 1;

  while (p1 >= 0 && p2 >= 0) {
    if (nums1[p1] > nums2[p2]) {
      nums1[p3] = nums1[p1];
      p1--;
    } else {
      nums1[p3] = nums2[p2];
      p2--;
    }
    p3--;
  }

  //When nums2 still has item left
  while (p2 >= 0) {
    nums1[p3] = nums2[p2];
    p3--;
    p2--;
  }
};
